TT-7903 DX-661 added otel in distributed tracing section

sredxny opened the pull request:
DX Ticket: https://tyktech.atlassian.net/browse/DX-661
---
### Preview Link
https://deploy-preview-3186--tyk-docs.netlify.app/docs/nightly/advanced-configuration/opentracing/

### Description
As we are about to release otel in tyk v5.2 we need to expose documentation of this new feature, how to enable, the config options and which data is added to the spans

Jira link: https://tyktech.atlassian.net/browse/TT-7903

#### Screenshots (if appropriate)

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR
<!-- Label your PR according to the type of changes that your code introduces. This ensures that we know how/when to publish the PR. These are the options:
- Fixing typo (please merge to production) - add the label `now`
- Documenting a new feature (please merge to production) - add the label `now`
- Documentation for future release (please do not merge to production) - add label `future release` and label of the release
- [Something else (please add if needs merging to production or not) - add label according to the above logic -->

sredxny commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+Choose between HTTP and gRPC for the backend connection by configuring the `exporter` field to "http" or "grpc".
+
+#### Further Configuration Details
+For more in-depth information on the configuration options available, please refer to our [OpenTelemetry Configuration Details Page](https://tyk.io/docs/tyk-oss-gateway/configuration/#opentelemetry).
```
This links should be updated once we re-generate and release the config options of otel

tbuchaillot commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+Alternatively, you can use the equivalent environment variable to enable OpenTelemetry.
+
+##### Step 2: Enable at API Level
+After enabling OpenTelemetry at the gateway level, activate it for the specific APIs you want to monitor. Edit the respective API definition to set `detailed_tracing` to `true`.
```
What `detailed_tracing ` do? How does an API trace looks with and without detailed_tracing enabled?

dcs3spp replied:
similar to comment above, what kind of information will detailed tracing log and maybe include an example of an API trace.

tbuchaillot commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+##### Step 2: Enable at API Level
+After enabling OpenTelemetry at the gateway level, activate it for the specific APIs you want to monitor. Edit the respective API definition to set `detailed_tracing` to `true`.
+
+#### Customizing Tracing
```
What's the customizable part of this?

tbuchaillot commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+- `tyk.api.name`: API name
+- `tyk.api.orgid`: Organization ID
+- `tyk.api.id`: API ID
+- `tyk.api.path`: API listen path
```
Please add as well resource attributes

tbuchaillot commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+
+#### Advanced OpenTelemetry Capabilities
+##### Context Propagation and Sampling
+Tyk supports context propagation and configurable sampling strategies ("AlwaysOn", "AlwaysOff", "TraceIDRatioBased").
```
How can you set different ratios? How does this affect the traces? More info

dcs3spp commented on tyk-docs/content/advanced-configuration/opentracing.md:
````diff
@@ -8,14 +8,71 @@ menu:
+}
+```
+
+Alternatively, you can use the equivalent environment variable to enable OpenTelemetry.
````
Name and link the equivalent environment variable here

dcs3spp commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+##### Tyk-Specific Attributes and Tags
+Tyk sets specific span attributes automatically:
+
+- `tyk.api.name`: API name
```
For consistent styling with other bullet points end the bullet points below with .
```suggestion
- `tyk.api.name`: API name.
```

dcs3spp commented on tyk-docs/content/advanced-configuration/opentracing.md:
```diff
@@ -8,14 +8,71 @@ menu:
+- `tyk.api.id`: API ID
+- `tyk.api.path`: API listen path
+
+When tagging is enabled in the API definition, tags are added as another span attribute, keyed as `tyk.api.tags`.
```
```suggestion
When tagging is enabled in the API definition, tags are added as another span attribute. These are keyed as `tyk.api.tags`.
```

dcs3spp reviewed (changes requested):
PR looks good to me @sredxny. I made some minor suggestions. Once review comments have been addressed we can merge the PR.

sredxny commented:
@tbuchaillot @dcs3spp lets close this, and review https://github.com/TykTechnologies/tyk-docs/pull/3212

There I addressed all your comments, and also contains the new structure of the distributed tracing section

sredxny commented:
closing in favor of: https://github.com/TykTechnologies/tyk-docs/pull/3212
